modules/sku/src/Plugin/facets/processor/HideTaxonomyNotInMenu.php

```diff
@@ -0,0 +1,143 @@
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\Language\LanguageManagerInterface $language_manager
+   *   The language manager.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, LanguageManagerInterface $language_manager, EntityTypeManagerInterface $entity_type_manager) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+
+    $this->languageManager = $language_manager;
+    $this->entityTypeManager = $entity_type_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('language_manager'),
+      $container->get('entity_type.manager')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function build(FacetInterface $facet, array $results) {
+    // Don't do anything if results set is empty. We can face this case
+    // when search results are empty.
+    if (empty($results)) {
+      return $results;
+    }
+
+    $source = $facet->getFacetSource();
+
+    // Support multiple entity types when using Search API.
+    if (($source instanceof SearchApiDisplay) &&
+      ($facet->getUseHierarchy())) {
+      $field_id = $facet->getFieldIdentifier();
+
+      // Load the index from the source, load the definition from the
+      // datasource.
+      /** @var \Drupal\facets\FacetSource\SearchApiFacetSourceInterface $source */
+      $index = $source->getIndex();
+      $field = $index->getField($field_id);
+
+      // Determine the target entity type.
+      $entity_type = $field->getDataDefinition()
+        ->getPropertyDefinition('entity')
+        ->getTargetDefinition()
+        ->getEntityTypeId();
+
+      // Process taxonomy terms & remove items not included in menu.
+      if ($entity_type == 'taxonomy_term') {
+        $ids = [];
+
+        /** @var \Drupal\facets\Result\ResultInterface $result */
+        foreach ($results as $delta => $result) {
+          $ids[$delta] = $result->getRawValue();
+        }
+
+        // Load all indexed entities of this type.
+        $entities = $this->entityTypeManager
+          ->getStorage($entity_type)
+          ->loadMultiple($ids);
+
+        // Loop over all results.
+        foreach ($results as $i => $result) {
+          $term = isset($entities[$ids[$i]]) ? $entities[$ids[$i]] : NULL;
+
+          // Display the term if included in menu and status is enabled.
+          if (($term instanceof TermInterface) &&
+            ($term->bundle() == 'acm_product_category') &&
```

Comment:
I think that should come from config `acm.connector. category_vid `

Reply:
PR Updated! Thanks!